handlers/auth.go

```diff
@@ -0,0 +1,112 @@
+package handlers
+
+import (
+	"encoding/json"
+	"log"
+	"net/http"
+	"time"
+
+	"github.com/Daniorocket/cookit-backend/lib"
+	"github.com/Daniorocket/cookit-backend/models"
+	uuid "github.com/satori/go.uuid"
+	"golang.org/x/crypto/bcrypt"
+)
+
+func (d *Handler) Login(w http.ResponseWriter, r *http.Request) {
+	login := &models.Login{}
+	err := json.NewDecoder(r.Body).Decode(login)
+	if err != nil {
+		log.Println("Error:", err)
+		if err := CreateApiResponse(w, nil, http.StatusBadRequest, "failed", "Failed to decode json"); err != nil {
+			log.Println("Error create Api response:", err)
+		}
+		return
+	}
+	passDB, err := models.GetPasswordByUsernameOrEmail(d.Client, d.DatabaseName, login.Username)
+	if err != nil {
+		log.Println("Error:", err)
+		if err := CreateApiResponse(w, nil, http.StatusBadRequest, "failed", "Invalid username or password"); err != nil {
```

Comment:
Może te logowanie błędu warto przenieść do tej funkcji `CreateApiResponse`, tam prawdopodobieństwo errora jest bardzo niskie, a musisz pisać 3 dodatkowe linijki zamiast 
```
log.Println(...)
CreateApiResponse(w, nil, http.StatusBadRequest, "failed", "Invalid username or password")
return
```

Reply:
Tak, właśnie tutaj miałem zagwostkę, w jaki sposób obsłużyć błędy przy tworzeniu odpowiedzi, a też to rozwiązanie wymagało dodatkowych linijek, które przy każdym wywołaniu musiałem powtórzyć.